src/amo/components/InstallWarning/index.js

```diff
@@ -0,0 +1,153 @@
+/* @flow */
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import type { AppState } from 'amo/store';
+import { hasAddonManager } from 'core/addonManager';
+import { ADDON_TYPE_EXTENSION, UNINSTALLED, UNKNOWN } from 'core/constants';
+import translate from 'core/i18n/translate';
+import log from 'core/logger';
+import tracking from 'core/tracking';
+import { isFirefox } from 'core/utils/compatibility';
+import { withExperiment } from 'core/withExperiment';
+import Notice from 'ui/components/Notice';
+import type { UserAgentInfoType } from 'core/reducers/api';
+import type { InstalledAddon } from 'core/reducers/installations';
+import type { AddonType } from 'core/types/addons';
+import type { I18nType } from 'core/types/i18n';
+import type { WithExperimentInjectedProps } from 'core/withExperiment';
+
+import './styles.scss';
+
+type Props = {|
+  addon: AddonType,
+|};
+
+type InternalProps = {|
+  ...Props,
+  ...WithExperimentInjectedProps,
+  _couldShowWarning?: () => boolean,
+  _hasAddonManager: typeof hasAddonManager,
+  _log: typeof log,
+  _tracking: typeof tracking,
+  i18n: I18nType,
+  installStatus: $PropertyType<InstalledAddon, 'status'>,
+  userAgentInfo: UserAgentInfoType,
+|};
+
+export const EXPERIMENT_CATEGORY_CLICK =
+  'AMO Install Button Warning Experiment - Click';
+export const EXPERIMENT_CATEGORY_DISPLAY =
+  'AMO Install Button Warning Experiment - Display';
+export const EXPERIMENT_ID = 'installButtonWarning';
+// We use dimension6 because that is a custom GA dimension added for this
+// particular experiment.
+export const INSTALL_WARNING_EXPERIMENT_DIMENSION = 'dimension6';
+export const VARIANT_INCLUDE_WARNING = 'includeWarning';
+export const VARIANT_EXCLUDE_WARNING = 'excludeWarning';
+
+export class InstallWarningBase extends React.Component<InternalProps> {
+  static defaultProps = {
+    _hasAddonManager: hasAddonManager,
+    _log: log,
+    _tracking: tracking,
+  };
+
+  couldShowWarning = () => {
+    const {
+      _couldShowWarning,
+      _hasAddonManager,
+      addon,
+      experimentIsEnabled,
+      installStatus,
+      userAgentInfo,
+    } = this.props;
+    return _couldShowWarning
+      ? _couldShowWarning()
```

Comment:
I am not sure to like this pattern. Could you file an issue so that we can discuss it and maybe revisit it later?

Reply:
I opened https://github.com/mozilla/addons-frontend/issues/8473.